tests/com/jvm/external/jar/AddJarManifestEntryTest.java

```diff
@@ -359,26 +391,50 @@ private Path createJarWithAutomaticModuleName(String name) throws IOException {
     return jar;
   }
 
-  private void createJar(Path outputTo, Manifest manifest, Map<String, String> pathToContents) throws IOException {
+  private void createJar(Path outputTo, Manifest manifest, Map<String, String> pathToContents)
+          throws IOException {
+    createJar(outputTo, manifest, pathToContents, false);
+  }
+
+  private void createJar(
+          Path outputTo,
+          Manifest manifest,
+          Map<String, String> pathToContents,
+          boolean setExtendedTimestamps)
+          throws IOException {
     try (OutputStream os = Files.newOutputStream(outputTo);
          ZipOutputStream zos = new JarOutputStream(os)) {
 
       if (manifest != null) {
         manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
         ZipEntry manifestEntry = new ZipEntry(JarFile.MANIFEST_NAME);
+
         zos.putNextEntry(manifestEntry);
         manifest.write(zos);
       }
 
       for (Map.Entry<String, String> entry : pathToContents.entrySet()) {
         ZipEntry ze = new ZipEntry(entry.getKey());
+        if (setExtendedTimestamps) {
+          ze.setLastModifiedTime(FileTime.from(Instant.now()));
+        }
         zos.putNextEntry(ze);
         zos.write(entry.getValue().getBytes(UTF_8));
         zos.closeEntry();
       }
     }
   }
 
+  private void writeJarInTimezone(Path inputJar, Path outputJar, String timeZone) throws IOException {
+    final TimeZone previousTimeZone = TimeZone.getDefault();
+    TimeZone.setDefault(TimeZone.getTimeZone(timeZone));
+    AddJarManifestEntry.main(new String[]{
```

Comment:
Please use 
```java
AddJarManifestEntry adder = new AddJarManifestEntry()
adder.setTimeZone(timezone);
adder.addEntryToManifest(...)
```

We modified the class to make this possible without needing to reset the default timezone

Reply:
I'm not sure how this would help. I can't pass or inject the time zone into `ZipEntry` class. Could you give me a more detailed example if you think that's the way to go?

Reply:
We can add a `TimeZone` as an argument in the `StableZipEntry`. Given that this is only needed for testing, and given that we're running the tests using Bazel's test runner, the modification of the default timezone for the test should be okay